spacy_llm/backends/dolly.py

```diff
@@ -79,8 +79,13 @@ def backend_dolly_hf(
         config = _compile_default_config()
 
     llm_pipeline = transformers.pipeline(model=model, **config)
+    return DollyBackend(llm_pipeline).query
 
-    def query(prompts: Iterable[str]) -> Iterable[str]:
-        return [llm_pipeline(pr)[0]["generated_text"] for pr in prompts]
 
-    return query
+class DollyBackend:
```

Comment:
Should we have a base backend to inherit from? Should `query` here just be `__call__`?

Reply:
Also will add docstrings and tests if we agree on this approach